### PR TITLE
Refactor ippool and securitypolicy service

### PR DESCRIPTION
**Pull request by seanpang-vmware**
1. Refactor ippool and security policy service
2. move some func from mediator to vpc service

### CODE REVIEW

File: pkg/nsx/services/mediator/mediator.go [hunk 1]
```diff
@@ -32,24 +32,6 @@ type ServiceMediator struct {
 	*node.NodeService
 }
```
Comment (timdengyun) [inline pkg/nsx/services/mediator/mediator.go:34]: do we still need ServiceMediator struct?

**Review by timdengyun — COMMENTED** [review]
SP parts looks ok.

File: pkg/nsx/services/mediator/mediator.go [hunk 2]
```diff
@@ -32,24 +32,6 @@ type ServiceMediator struct {
...
-			log.Error(err, "Failed to get vpc info from vpc path", "vpc path", *v.Path)
-		}
-		VPCInfoList = append(VPCInfoList, vpcResourceInfo)
-	}
-	return VPCInfoList
-}
-
 // This method is used for subnet service since vpc network config contains default subnet size
 // and default subnet access mode.
 func (m *ServiceMediator) GetVPCNetworkConfigByNamespace(ns string) *common.VPCNetworkConfigInfo {
```
Comment (timdengyun) [inline pkg/nsx/services/mediator/mediator.go:37]: do we still need this func in mediator?

File: pkg/nsx/services/mediator/mediator.go (hunk 1 above)
Comment (timdengyun) [reply pkg/nsx/services/mediator/mediator.go:34]: it looks like we can remove this mediator file.

**Comment by timdengyun** [comment]
since the PR  https://github.com/vmware-tanzu/nsx-operator/pull/392 merged, please rebase and fix conflicts

**Review by zhengxiexie — COMMENTED** [review]
IPPool part looks great!

File: pkg/nsx/services/mediator/mediator.go (hunk 1 above)
Comment (seanpang-vmware) [reply pkg/nsx/services/mediator/mediator.go:34]: not now, there is another patch refactoring subnet and subnetset, then we can remove them

File: pkg/nsx/services/mediator/mediator.go (hunk 2 above)
Comment (seanpang-vmware) [reply pkg/nsx/services/mediator/mediator.go:37]: it is removed.

**Comment by seanpang-vmware** [comment]
> since the PR #392 merged, please rebase and fix conflicts

done

**Review by timdengyun — APPROVED** [review]
LGTM

**Pull request merged**